File: GreenSignal/Data/Models/IncidentReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Data.Models
{
    public class IncidentReport
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public string SerialNumber { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public Guid InspectorId { get; set; }
        public Inspector Inspector { get; set; }

        [Required]
        //Дата создания акта
        public DateTime CreatedAt { get; set; }

        [Required]
        //Дата оформления акта
        public DateTime ManualDate { get; set; }

        [Required]
        public DateTime StartOfInspection { get; set; }

        [Required]
        public DateTime EndOfInspection { get; set; }

        public Guid? IncidentId { get; set; }
        public Incident? Incident { get; set; }

        public Guid? LocationId { get; set; }
        public Location? Location { get; set; }

        [Required]
        public IncidentReportStatus Status { get; set; }

        [Required]
        public IncidentReportKind Kind { get; set; }

        [Required]
        public string Address { get; set; }
        
        [Required]
        public double Lat { get; set; }
        [Required]
        public double Lng { get; set; }

        [Required]
        public int AttributesVersion { get; set; }

        public IEnumerable<IncidentReportAttachment> IncidentReportAttachments { get; set; }
        public IEnumerable<IncidentReportAttribute> IncidentReportAttributes { get; set; }
    }


    public enum IncidentReportKind
    {
        [Display(Name = "Загрязнение воздуха")]
        AirPollution,
        [Display(Name = "Загрязнение почвы")]
        SoilPollution,
        [Display(Name = "Добыча недр")]
        Excavation,
        [Display(Name = "Свалка")]
        Dump,
        [Display(Name = "Вырубка деревьев")]
        TreeCutting,
        [Display(Name = "Радиация")]
        Radiation
    }

    public enum IncidentReportStatus
    {
        Draft = 0,
        Sent = 100,
        Completed_successfuly = 200,
        Completed_unsucessful = 300,
        Archived = 400
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/IncidentReport.kt
package com.example.greensignal.domain.model.response

import com.example.greensignal.data.remote.dto.response.IncidentReportDto
import com.example.greensignal.data.remote.dto.response.IncidentReportKind
import com.example.greensignal.data.remote.dto.response.IncidentReportStatus
import java.util.Date

data class IncidentReport(
    val id: String = "",
    val serialNumber: String = "",
    val address: String = "",
    val inspectorId: String = "",
    val inspector: Inspector = Inspector(),
    val description: String = "",
    val createdAt: Date = Date(),
    val manualDate: Date = Date(),
    val startOfInspection: Date = Date(),
    val endOfInspection: Date = Date(),
    val incidentId: String? = null,
    val incident: Incident? = null,
    val locationId: String? = null,
    val location: Location? = null,
    val status: IncidentReportStatus = IncidentReportStatus.Draft,
    val kind: IncidentReportKind = IncidentReportKind.Dump,
    val lat: Double = 0.0,
    val lng: Double = 0.0,
    val attributesVersion: Int = 0,
    val incidentReportAttachements: MutableList<IncidentReportAttachment> = mutableListOf(),
    val incidentReportAttributes: MutableList<IncidentReportAttribute> = mutableListOf()
)

fun IncidentReportDto.toIncidentReport(): IncidentReport {
    return IncidentReport(
        id = id,
        serialNumber = serialNumber,
        address = address,
        description = description,
        inspectorId = inspectorId,
        inspector = inspector.toInspectorAccount(),
        createdAt = createdAt,
        manualDate = manualDate,
        startOfInspection = startOfInspection,
        endOfInspection = endOfInspection,
        incidentId = incidentId,
        incident = incident?.toIncident(),
        locationId = locationId,
        location = location?.toLocation(),
        status = status,
        kind = kind,
        lat = lat,
        lng = lng,
        attributesVersion = attributesVersion,
        incidentReportAttachements = incidentReportAttachments.map { it.toIncidentReportAttachment() }.toMutableList(),
        incidentReportAttributes = incidentReportAttributes.map { it.toIncidentReportAttribute() }.toMutableList()
    )
}